core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -141,7 +141,9 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId))
+      // Tombstone does not care about message magic value, simply use magic value 0.
+      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId),
+        timestamp = Message.NoTimestamp, magicValue = Message.MagicValue_V0)
```

Comment:
Hi Jun, currently we always do re-compression when compacting the log, even if message format v1 is used. Do you mean we should change that so if a compressed message set does not change after compaction we simply write the original message set back? If we do that (and we probably should), it seems it does not matter whether message format v0 or v1 is used, because it only depends on whether there is message in the message set got compacted out or not.